site/src/sphinx/server-annotated-service.rst

```diff
@@ -0,0 +1,718 @@
+    }
+
+.. code-block:: java
+
+    enum CaseSensitive {
+        ALPHA, alpha
+    }
+
+.. code-block:: java
+
+    class MyAnnotatedServiceClass {
+
+        @Get("/hello1/{there}")
+        public HttpResponse hello1(@Param("there") CaseInsensitive there) {
+            // 'there' is converted in a case-insensitive way.
+        }
+
+        @Get("/hello2/{there}")
+        public HttpResponse hello2(@Param("there") CaseSensitive there) {
+            // 'there' must be converted in a case-sensitive way.
+            // So 'ALPHA' and 'alpha' are only acceptable.
+        }
+    }
+
+When the value of `@Param`_ annotation is not shown in the path pattern, it will be handled as a parameter
+name of the query string of the request. If you have a service class like the example below and a user sends an
+HTTP GET request with URI of ``/hello1?name=armeria``, the service method will get ``armeria`` as the value
+of parameter ``name``. If there is no parameter named ``name`` in the query string, the parameter ``name``
+of the method would be ``null``. If you want to avoid ``null`` in this case, you can use `@Default`_
+annotation or ``Optional<?>`` class, e.g. ``hello2`` and ``hello3`` method in the example, respectively.
+
+.. code-block:: java
+
+    class MyAnnotatedServiceClass {
+
+        @Get("/hello1")
+        public HttpResponse hello1(@Param("name") String name) { ... }
+
+        @Get("/hello2")
+        public HttpResponse hello2(@Param("name") @Default("armeria") String name) { ... }
+
+        @Get("/hello3")
+        public HttpResponse hello3(@Param("name") Optional<String> name) {
+            String clientName = name.orElse("armeria");
+            ...
+        }
+    }
+
+Armeria also provides `@Header`_ annotation to inject an HTTP header value into a parameter. The parameter
+annotated with `@Header`_ is also specified as one of the built-in types as follows. `@Default`_ and
+``Optional<?>`` are also supported.
+
+.. code-block:: java
+
+    class MyAnnotatedServiceClass {
+
+        @Get("/hello1")
+        public HttpResponse hello1(@Header("Authorization") String auth) { ... }
+
+        @Post("/hello2")
+        public HttpResponse hello2(@Param("Content-Length") long contentLength) { ... }
+    }
+
+Moreover, the following classes are automatically injected into your parameter when you specify it on the
+parameter list of your method.
+
+- RequestContext_
+- ServiceRequestContext_
+- Request_
+- HttpRequest_
+- AggregatedHttpMessage_
+- HttpParameters_
+
+.. code-block:: java
+
+    class MyAnnotatedServiceClass {
+
+        @Get("/hello1")
+        public HttpResponse hello1(ServiceRequestContext ctx, HttpRequest req) {
+            // Use the context and request inside a method.
+        }
+
+        @Post("/hello2")
+        public HttpResponse hello2(AggregatedHttpMessage aggregatedMessage) {
+            // Armeria aggregates every message received, then call this method with an aggregated message.
```

Comment:
every message received, then call -> the received HttpRequest and calls .. with the aggregated request.